src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -325,3 +333,76 @@ func (c controller) sendCompleteInstallation(isSuccess bool, errorInfo string) {
 	}
 	c.log.Infof("Done complete installation step")
 }
+
+// get pod logs,
+// write tar.gz to pipe in a routine
+// upload tar.gz from pipe to assisted service.
+// close read and write pipes
+func (c controller) uploadPodLogs(podName string, namespace string) error {
+	c.log.Infof("Uploading logs for %s in %s", podName, namespace)
+	podLogs, err := c.kc.GetPodLogs(namespace, podName, logsSinceInSec)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get logs of pod %s", podName)
+	}
+	if podLogs == "" {
+		return errors.Errorf("no logs of %s were found", podName)
+	}
+	pr, pw := io.Pipe()
+	defer pr.Close()
+
+	go func() {
+		defer pw.Close()
+		err = utils.AddStringToTarGz(pw, podLogs, fmt.Sprintf("%s.logs", podName))
+		if err != nil {
+			c.log.WithError(err).Warnf("Failed to upload")
+		}
+	}()
+	// if error will occur in goroutine above
+	//it will close writer and upload will fail
+	err = c.ic.UploadLogs(c.ClusterID, models.LogsTypeController, pr)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to upload logs")
+	}
+	return nil
+}
+
+// Uploading logs every 5 minutes
+// We will take logs of assisted controller and upload them to assisted-service
+// by creating tar gz of them.
+func (c *controller) UploadControllerLogs(ctx context.Context, wg *sync.WaitGroup) {
```

Comment:
Make a generic function getting job-name as a variable
Fix the logs accordingly.
```Failed to get %s..```
```Didn't find %s...```

Reply:
It can't be generic. It is relevant to specific pod. I know that in assisted-installer namespace we have only one.

Reply:
to have generic function that does getPods? i have it in client, i still will need to check it's error and to verify i got pods at all.